controllers/comments.controller.js

```diff
@@ -1,40 +1,63 @@
 const {
   selectCommentsByArticleId,
   addComment,
+  removeSelectedComment,
 } = require("../models/comments.model")
 
-const checkCommentsExists = require("../models/checkIfExists")
-const checkArticleExists = require("../models/checkIfExists")
+const checkExists = require("../models/checkIfExists")
 
 function getCommentsByArticleId(req, res, next) {
   const { article_id } = req.params
-  const promises = [selectCommentsByArticleId(article_id)]
-  if (article_id) {
-    promises.push(checkArticleExists(article_id))
-  }
-  Promise.all(promises)
-    .then(([comments]) => { 
-        console.log(comments);
-             
+  selectCommentsByArticleId(article_id)
+    .then((comments) => {
       res.status(200).send({ comments })
     })
     .catch((err) => {
       next(err)
     })
 }
 
-function postCommentOnArticle(req, res, next) { 
+function postCommentOnArticle(req, res, next) {
   const { article_id } = req.params
   const { username, body } = req.body
   addComment(article_id, username, body)
-  .then((comment) => {
+    .then((comment) => {
       res.status(201).send({ comment })
     })
     .catch((err) => {
       next(err)
     })
 }
+// function deleteComment(req, res, next) {
+//   const { comment_id } = req.params
+//   const promises = [removeSelectedComment(comment_id)]
+//   if (comment_id) {
+//     promises.push(checkExists("comments", "comment_id", comment_id))
+//   }
+//   Promise.all(promises)
+//     .then(() => {
+//       res.status(204).send()
+//     })
+//     .catch((err) => {
+//       next(err)
+//     })
+// }
 
+//will return later and try and fix not working properly only works occasionally unsure why
 
-module.exports = { getCommentsByArticleId, postCommentOnArticle }
+function deleteComment(req, res, next) {
+  const { comment_id } = req.params
+  const promises = [
+    checkExists("comments", "comment_id", comment_id),
+    removeSelectedComment(comment_id),
+  ]
```

Comment:
This works the way you've done it, but it isn't actually necessary to call the checkExists model here. You can throw a Promise.reject from your model if the comment doesn't exist.
Remember, we only want to be checking if the comment exists in a separate model if we want to send back an empty array. For this endpoint however, if the comment doesn't exist, we're always going to throw an error 👍